File: presentation/src/main/java/com/example/presentation/utils/scheduler/IBaseSchedulerProvider.java
package com.example.presentation.utils.scheduler;

import org.jetbrains.annotations.NotNull;

import io.reactivex.Scheduler;

/**
 * Интерфейс-обертка над планировщиками RxJava
 */
public interface IBaseSchedulerProvider {

    /**
     * Обертка над io планировщиком
     */
    @NotNull
    Scheduler io();

    /**
     * Обертка над computation планировщиком
     */
    @NotNull
    Scheduler computation();

    /**
     * Обертка над планировщиком главного потока
     */
    @NotNull
    Scheduler mainThread();
}

File: presentation/src/main/java/com/example/presentation/utils/scheduler/IBaseSchedulerProvider.kt
package com.example.presentation.utils.scheduler

import io.reactivex.Scheduler

/**
 * Интерфейс-обертка над планировщиками RxJava
 */
interface IBaseSchedulerProvider {
    /**
     * Обертка над io планировщиком
     */
    fun io(): Scheduler

    /**
     * Обертка над computation планировщиком
     */
    fun computation(): Scheduler

    /**
     * Обертка над планировщиком главного потока
     */
    fun mainThread(): Scheduler
}